src/main.rs

```diff
@@ -333,52 +338,56 @@ mod tests {
 
     #[test]
     fn test_translate_atg() {
-        assert_eq!(translate("ATG", 1), 'M');
+        assert!(matches!(translate(b"ATG", 1), Ok('M')));
     }
 
     #[test]
     fn test_translate_atg_as_start() {
-        assert_eq!(translate("ATG", 0), 'M');
+        assert!(matches!(translate(b"ATG", 0), Ok('M')));
     }
 
     #[test]
     fn test_translate_gtg() {
-        assert_eq!(translate("GTG", 1), 'V');
+        assert!(matches!(translate(b"GTG", 1), Ok('V')));
     }
 
     #[test]
     fn test_translate_gtg_as_start() {
-        assert_eq!(translate("GTG", 0), 'M');
+        assert!(matches!(translate(b"GTG", 0), Ok('M')));
     }
 
     #[test]
     fn test_translate_tag() {
-        assert_eq!(translate("TAG", 1), '*');
+        assert!(matches!(translate(b"TAG", 1), Ok('*')));
     }
 
     #[test]
     fn test_translate_ttt() {
-        assert_eq!(translate("TTT", 1), 'F');
+        assert!(matches!(translate(b"TTT", 1), Ok('F')));
     }
 
     #[test]
     fn test_one_to_three_translate() {
-        assert_eq!(three_letter_code(translate("ATG", 0)), "Met");
+        assert_eq!(
+            three_letter_code(translate(b"ATG", 0).unwrap()).unwrap(),
```

Comment:
(It was harder to do the `matches!` macro here, so I ended up doing this double `.unwrap()`. Shame on me)